### PR TITLE
[WIP] feat: PREVIEW Page and other components

### PR DESCRIPTION
**Pull request by kaljarv**
This is a preview dump of the `Page` and other components. Before merging the monolithic commit will be split into multiple commits and possibly PRs.

## WHY:

Fixes 
- #19
- #60 
- #109 
- #214 

### What has been changed (if possible, add screenshots, gifs, etc. )

A shared `Page` component and subcomponents, such as `BasicPage` and `FrontPage` have been created that can be used to easily create pages that use a common layout.

In addition, a number of other components are defined for icons, buttons, navigation etc.

As additional quick fixes, the candidates' answers are now included in the placeholder `CandidateOpinions` component.

## Check off each of the following tasks as they are completed

- [ ] I have reviewed the changes myself in this PR. Please check the [self-review document](https://github.com/OpenVAA/voting-advice-application/blob/main/docs/contributing/self-review.md)
- [ ] I have added or edited unit tests.
- [ ] I have run the unit tests successfully.
- [ ] I have tested this change on my own device.
- [ ] I have tested this change on other devices (Using Browserstack is recommended).
- [ ] I have tested my changes using the [WAVE extension](https://wave.webaim.org/extension/)
- [ ] I have added documentation where necessary.
- [ ] Is there an existing issue linked to this PR?

**Clean up your git commit history before submitting the pull request!**

### CODE REVIEW

**Comment by kaljarv** [comment]
I’m now always applying the `$$restProps` to the outermost HTML element or base Svelte component in case of a derived component. To enable supplying additional classes , those are plucked from the `$$restProps` and appended to the classes needed by the component. You can see that in action if you look at basically any of the components.

The best place to start is maybe to have a look at `$lib/components/shared/page` and then `$lib/components/basicPage`.

Note that there's some copy-pasting between `BasicPage` and  `SingleCardPage`, which could be gotten rid of by creating a common parent for these.

NB. The `/candidates` and `/parties` routes have not been updated and do not work properly now.

File: frontend/src/lib/components/heroEmoji/HeroEmoji.svelte [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  type $$Props = HeroEmojiProps;
+
+  // Merge the necessary classes to possible extra classes defined in the props
+  $$restProps.class = `text-center font-emoji text-[6.5rem] leading-[6.5rem] ${
+    $$restProps.class ?? ''
+  }`;
+</script>
+
+<!--
```
Comment (jaguarfi) [inline frontend/src/lib/components/heroEmoji/HeroEmoji.svelte:12]: are these comments used for some autogenerated documentation?

File: frontend/src/lib/components/icons/AddToListIcon.svelte [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<script lang="ts">
```
Comment (jaguarfi) [inline frontend/src/lib/components/icons/AddToListIcon.svelte:1]: All the icons contain the same code. Should it be better to create a generic Icon and pass the name of the icon we want to render?. We could use a types list to provide some intellisense and allow people to know the available icons

**Review by jaguarfi — COMMENTED** [review]
I reviewed some changes. But it is better to split this PR into several ones. Otherwise, all the comments will get lost. i suggest the following PRs:
- one for navigation menu
- one for templates/pages
- one for the adjustment to the candidateDetailsCard
- one for adding the CandidateOpinions
- one or many for small adjustments such as the EntityCard adjustments, Question adjustments, utils, logger and types adjustments and other small adjustments.
- one for the HeroIcon/HeroEmoji
- one for the button
- one for all the icons

Basically, I may suggest creating one PR per issue. If the issue does not exist, we can create it and then create the PR. This may help us to understand the reason for the changes.

The UI looks pretty nice. But we need to split this now to start merging small changes and move all the required changes forward.

File: frontend/README.md [hunk 3]
```diff
@@ -72,3 +72,49 @@ you develop a new component, be sure to comply with the [Accessibility Guideline
 - [Svelte](https://svelte.dev/) - Svelte is a radical new approach to building user interfaces.
 - [SvelteKit](https://kit.svelte.dev/) - SvelteKit is a framework for building web applications of all sizes, with a
   beautiful development experience and flexible, low-level APIs.
+
+## Styling
```
[the remark this answers is not in the document]
Comment (kaljarv) [reply frontend/README.md:76]: I think that file is intended for coding style but we can definitively create another file using the frontend styles if we don't want to bloat the README.

File: frontend/README.md [hunk 4]
```diff
@@ -72,3 +72,49 @@ you develop a new component, be sure to comply with the [Accessibility Guideline
...
+|  <div style="background: #ffffff; outline: 1px solid #666666; outline-offset: -1px; width: 1.5rem; height: 1.5rem;"/> | `base-100`      | Default background                                                    |
+|  <div style="background: #a82525; width: 1.5rem; height: 1.5rem;"/>                                                   | `base-200`      | Slightly less prominent shaded background                             |
+|  <div style="background: #d1ebee; width: 1.5rem; height: 1.5rem;"/>                                                   | `base-300`      | Default prominent background                                          |
+|  <div style="background: #ffffff; outline: 1px solid #2546a8; outline-offset: -1px; width: 1.5rem; height: 1.5rem;"/> | primary-content | Text on `bg-primary`. Each colour has its associated `content` colour |
+
+### Default styling
+
+See `app.css` for some styling defaults that are set throughout the app.
+
+### OSX: Test locally on an iPhone simulator
```
[the remark this answers is not in the document]
Comment (kaljarv) [reply frontend/README.md:118]: It was mainly a note for self, so I'll just remove it.

File: frontend/src/lib/components/appLogo/AppLogo.type.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import type {HTMLAttributes} from 'svelte/elements';
+export interface AppLogoProps extends HTMLAttributes<HTMLElementTagNameMap['div']> {
+  /**
+   * If `true`, the light and dark versions of the logo will be reversed.
+   * Set to `true` if using the logo on a dark background.
+   *
+   * @default `false`
+   */
+  inverse?: boolean | undefined | null;
```
[the remark this answers is not in the document]
Comment (kaljarv) [reply frontend/src/lib/components/appLogo/AppLogo.type.ts:9]: Okay! I'll remove the unnecessary `undefined`s everywhere, but wasn't there a discussion about this earlier when dealing with `EntityCard` or smth? It seems there we still have `foo?: string | undefined | null` for aria attributes.

File: frontend/src/lib/components/appNav/AppNav.svelte [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      </a>
+    </li>
+    <li>
+      <a href="/questions" class="menu-item">
+        <OpinionIcon aria-hidden="true" />
+        {$page.data.appLabels.actionLabels.opinions}
+      </a>
+    </li>
+    <li>
+      <a href="/results" class="menu-item">
```
[the remark this answers is not in the document]
Comment (kaljarv) [reply frontend/src/lib/components/appNav/AppNav.svelte:49]: I'll add it! And in any case, more logic like that will be needed in the future I think.

File: frontend/src/lib/components/frontPage/FrontPage.svelte [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  $$restProps.headerClass = `!absolute w-full bg-transparent ${$$restProps.headerClass ?? ''}`;
+  $$restProps.mainWrapperClass = `!p-0 ${$$restProps.mainWrapperClass ?? ''}`;
+  $$restProps.navLabel ??= $_('header.navLabel');
+</script>
+
+<Page {title} {...$$restProps}>
+  <!-- Header -->
+  <svelte:fragment slot="navOpen">
+    <MenuIcon />
+    <AppLogo />
```
[the remark this answers is not in the document]
Comment (kaljarv) [reply frontend/src/lib/components/frontPage/FrontPage.svelte:25]: Yeah, that would be the standard functionality. However, in this case it might be confusing to the user whether the menu button is its separate button or part of the logo. As opening the menu would most likely be the more useful interaction here, I'd keep it this way. The home page can be anyway reached via a second click after the menu is open.

File: frontend/src/lib/components/singleCardPage/SingleCardPage.svelte [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+  $$restProps.headerClass = `bg-base-300 ${$$restProps.headerClass ?? ''}`;
+  $$restProps.asideClass = `text-secondary text-center mb-xl ${$$restProps.asideClass ?? ''}`;
+  $$restProps.navLabel ??= $_('header.navLabel');
+  // Explicitly hide aside if it's not supplied. We cannot wrap it in an if
+  // block below
+  if (!$$slots.aside) {
+    $$restProps.hideSlots ??= ['aside'];
+  }
+
+  // TODO: This template has a bit of code copy-pasted from `BasicPage.svelte`.
```
[the remark this answers is not in the document]
Comment (kaljarv) [reply frontend/src/lib/components/singleCardPage/SingleCardPage.svelte:25]: I'll check that. The other option is to make a common ancestor for both in order to copy less code. This single card template will be used on a few routes, so it makes sense to have it as a template.

File: frontend/src/lib/components/heroEmoji/HeroEmoji.svelte (hunk 1 above)
Comment (kaljarv) [reply frontend/src/lib/components/heroEmoji/HeroEmoji.svelte:12]: Yes they are, see https://svelte.dev/docs/faq#how-do-i-document-my-components

File: frontend/src/lib/components/icons/AddToListIcon.svelte (hunk 2 above)
Comment (kaljarv) [reply frontend/src/lib/components/icons/AddToListIcon.svelte:1]: There are some drawbacks to that, but we can go either way. See discussion in the new PR #234 

The code is not totally identical for each of them, though. What varies are:

- `title`
- `viewBox` (although it's the same in most of them)
- whether to apply `color` to `fill` or `stroke` (although in the current batch, it's always fill)
- the SVG contents, naturally

This approach was modelled after https://github.com/ramiroaisen/svelte-material-icons/tree/master/svelte-material-icons

File: frontend/src/lib/components/appLogo/AppLogo.type.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (jaguarfi) [reply frontend/src/lib/components/appLogo/AppLogo.type.ts:9]: oh yeah, I think we wanted to stick to Svelte's `AriaAttributes` way of defining the types.

[the remark this answers is not in the document]
Comment (kaljarv) [reply frontend/src/lib/components/appLogo/AppLogo.type.ts:9]: Okay, I guess they have a reason for that :)
